pallets/ajuna-awesome-avatars/src/types/avatar/versions/v2/avatar_utils.rs

```diff
@@ -1076,6 +1080,18 @@ where
 		(match_count > 0 && (((match_count * 2) + mirror_count) >= 6)).then_some(matches)
 	}
 
+	/// Matches two progress arrays, using `array_2` to upgrade `array_1`.
+	///
+	/// This method return the indexes that matched and the indexes that are considered mirrors.
+	///
+	/// * Matches are indexes that can be increased to the next rarity level.
+	/// * Mirrors are indexes that have been maxed and are of the same variation between both
+	///   arrays.
+	///
+	/// `rarity_level` is used to determine the maximum rarity for which a direct byte match is not
+	/// necessary for a matching to occur, the higher the value, the easier it is to generate
+	/// matches between two arrays, while at the minimum level (which is 0) only direct byte matches
+	/// or special variations will generate a match.
```

Comment:
My brain's sentence cache run out: XD

```suggestion
	/// `rarity_level` is used to determine the maximum rarity for which a direct byte match is not
	/// necessary for a matching to occur. The higher the value, the easier it is to generate
	/// matches between two arrays. At the minimum level (which is 0) only direct byte matches
	/// or special variations will generate a match.
```